### PR TITLE
feat: adding signin with password

### PR DESCRIPTION
**Pull request by katsumeshi**
try to restart #47
i would like to take this feature.

### CODE REVIEW

**Comment by katsumeshi** [comment]
any comments?

File: src/index.js [hunk 1]
```diff
@@ -183,9 +219,16 @@ export default class Firedux {
   }
   logout () {
     const { dispatch } = this
+
     return new Promise((resolve, reject) => {
       dispatch({type: 'FIREBASE_LOGOUT_ATTEMPT'})
-      this.ref.unauth()
+      if (this.v3) {
+        this.auth().signOut()
+        .then(() => resolve(), (error) => reject(error))
```
Comment (adjohnson916) [inline src/index.js]: Could shorten to `.then(resolve, reject)`.

File: src/index.js [hunk 2]
```diff
@@ -153,23 +177,35 @@ export default class Firedux {
...
         that.authData = authData
         dispatch({type: 'FIREBASE_LOGIN', authData: authData, error: error})
         resolve(authData)
       }
 
       try {
-        if (credentials.token) {
+        if (this.v3) {
+          // need to ignore redux
+          if (_.isFunction(credentials)) return null
```
Comment (adjohnson916) [inline src/index.js]: Why would `credentials` be a function? It's passed in by the user but we don't expect a function. Returning `null` here will cause the promise never to settle - probably not what we want.

File: src/index.js [hunk 3]
```diff
@@ -183,9 +219,16 @@ export default class Firedux {
...
     const { dispatch } = this
+
     return new Promise((resolve, reject) => {
       dispatch({type: 'FIREBASE_LOGOUT_ATTEMPT'})
-      this.ref.unauth()
+      if (this.v3) {
+        this.auth().signOut()
+        .then(() => resolve(), (error) => reject(error))
+      } else {
+        this.ref.unauth() // no callbacks for old firebase :(
```
Comment (adjohnson916) [inline src/index.js]: Should we call `resolve()` here anyway so at least the promise settles?

File: src/index.js (hunk 2 above)
Comment (katsumeshi) [inline src/index.js]: yeah, I agree with that. Credentials are supposed to be object, so we could remove the line , and we might be better to have conditional instead whether credentials contain email and password.probably, will reject if doesn't contents those.

File: src/index.js (hunk 1 above)
Comment (katsumeshi) [inline src/index.js]: i'm considering those things with unauth placed below. i'll update with PR. would be glad to review it after that. thanks

File: src/index.js (hunk 3 above)
Comment (katsumeshi) [inline src/index.js]: > remove the dist builds from the PR or regenerate for a clean merge?
I got it.

anyway, i'm bit confused about testing. i've tried to add the test case for logout to make sure working properly, however i got an error which is
> Your API key is invalid, please check you have copied it correctly.

should i run the npm like this?
`npm test FIREBASE_API_KEY=something api key`
or just put the apiKey in ref.js?

**Comment by katsumeshi** [comment]
> remove the dist builds from the PR or regenerate for a clean merge?
I got it.

anyway, i'm bit confused about testing. i've tried to add the test case for logout to make sure working properly, however i got an error which is
> Your API key is invalid, please check you have copied it correctly.

for now, i've put temporarily my own api key into ref.js

should i run the npm like this?
`npm test FIREBASE_API_KEY="something api key"`
or just put the key in ref.js?

**Comment by adjohnson916** [comment]
@katsumeshi Can you rebase so 9acd0904 commit message has `fix:` prefix? All commits should follow conventional commit standard.

**Comment by adjohnson916** [comment]
@katsumeshi Also we should rebase reword so 9acd0904c7de7baf374e5aef3a8420d63cd154be commit message follows AngularJS's conventions (e.g. `fix: ` prefix).

**Comment by katsumeshi** [comment]
@adjohnson916 I think it's changed to proper commit message. let me know if there is any something wrong. Thank you

**Comment by adjohnson916** [comment]
@katsumeshi I have merged. I'll need to do another release at some point. I'd like to consider adding some tests first. Are you interested?

**Comment by katsumeshi** [comment]
@adjohnson916 Appreciate it . I'm glad to merge this branch. Yeah, I'd love to help here out whatever I can, let me know what you want me to do.

**Comment by adjohnson916** [comment]
@katsumeshi @DarrylD I've published to npm as 1.1.0.

**Pull request merged**